Refactor domains cli to remove model call from regex

mjankowski opened the pull request:
Solves brakeman warning.

mjankowski committed:
Refactor domains cli to remove model call from regex

ClearlyClaire reviewed (commented):
I'm a bit split on this change, I don't think it makes it especially more readable, and if the aim is just to remove the brakeman warning, this sounds like “obfuscation”.

mjankowski commented:
Updated again to use the regexp you suggested, remove one of the methods I had added, rename the other one, and use the DomainBlock scope created by the enum.
